packages/astro/src/cli/index.ts

```diff
@@ -71,6 +74,62 @@ async function printVersion() {
 	console.log(`  ${colors.bgGreen(colors.black(` astro `))} ${colors.green(`v${ASTRO_VERSION}`)}`);
 }
 
+async function printInfo({
+	cwd,
+	flags,
+	logging,
+}: {
+	cwd?: string;
+	flags?: Flags;
+	logging: LogOptions;
+}) {
+	const whichPm = await import('which-pm');
+	const packageManager = await whichPm.default(process.cwd());
+	let adapter = "Couldn't determine.";
+	const integrations = [];
+
+	function isIntegration(integration: any): integration is AstroIntegration {
+		if (typeof integration === 'object') {
+			return true;
+		}
+		return false;
+	}
+
+	const MAX_PADDING = 25;
+	function printRow(label: string, value: string) {
+		const padding = MAX_PADDING - label.length;
+		console.log(`${colors.bold(label)}` + ' '.repeat(padding) + `${colors.green(value)}`);
+	}
+
+	try {
+		const { userConfig } = await openConfig({
+			cwd,
+			flags,
+			cmd: 'info',
+			logging,
+		});
+		if (userConfig.adapter?.name) {
+			adapter = userConfig.adapter.name;
+		}
+		if (userConfig.integrations) {
+			for (const integration of userConfig.integrations) {
+				if (isIntegration(integration)) {
+					integrations.push(integration.name);
+				}
```

Comment:
I believe this will run into the same bug I just fixed in #7429 — some integrations are `AstroIntegration[]` not `AstroIntegration`. Probably need to update the `isIntegration` helper to assert `is AstroIntegration | AstroIntegration[]` and then act accordingly?

Reply:
Good catch! Last commit should handle that